cmd/contour/contour.go

```diff
@@ -181,7 +187,35 @@ func main() {
 		et := &contour.EndpointsTranslator{
 			FieldLogger: log.WithField("context", "endpointstranslator"),
 		}
-		k8s.WatchEndpoints(&g, client, wl, et)
+		coreInformers.Core().V1().Endpoints().Informer().AddEventHandler(et)
+
+		g.Add(func(stop <-chan struct{}) error {
+			log := log.WithField("context", "coreinformers")
+
+			log.Println("waiting for cache sync from core informers")
+			coreInformers.WaitForCacheSync(stop)
+
+			log.Println("starting core shared informer")
+			defer log.Println("stopping core shared informer")
+			coreInformers.Start(stop)
+
+			<-stop
+			return nil
+		})
+
+		g.Add(func(stop <-chan struct{}) error {
+			log := log.WithField("context", "contourinformers")
```

Comment:
same as above.

It might be worth breaking this out into a helper function that takes an interface with two methods, `WaitForCacheSync` and `Start`.

Reply:
Updated. This also required the helper func to accept the stop channel and the logger.